app/src/pages/playground/PlaygroundTemplate.tsx

```diff
@@ -117,3 +133,50 @@ function DeleteButton(props: PlaygroundInstanceProps) {
     </TooltipTrigger>
   );
 }
+
+type SaveButtonProps = {
+  instanceId: number;
+  setDialog: (dialog: React.ReactNode) => void;
+};
+
+function SaveButton({ instanceId, setDialog }: SaveButtonProps) {
+  const instance = usePlaygroundContext((state) =>
+    state.instances.find((instance) => instance.id === instanceId)
+  );
+  if (!instance) {
+    throw new Error(`Instance ${instanceId} not found`);
+  }
+  const prompt = instance.prompt;
+  const onSubmit: SavePromptSubmitHandler = useCallback(
+    (params) => {
+      // eslint-disable-next-line no-console
+      console.log("saving prompt", instanceId, params);
+    },
+    [instanceId]
+  );
+  const onSave = () => {
+    setDialog(
+      <Dialog title={prompt?.id ? "Update Prompt" : "Save Prompt"}>
```

Comment:
Will likely take this one step further when we are actually getting the network involved by updating the title with the prompt's name